demos/build_demos.sh

```diff
@@ -82,6 +88,15 @@ fi
 mkdir -p "$build_dir"
 
 (cd "$build_dir" && cmake -DCMAKE_BUILD_TYPE=Release "${extra_cmake_opts[@]}" "$DEMOS_PATH")
-cmake --build "$build_dir" -- "$NUM_THREADS"
+
+if [ -z "$build_targets" ]; then
```

Comment:
`$build_targets` substitutes to just the first element of `build_targets`, therefore this doesn't actually check if the array is empty. Use `${#build_targets[@]} -eq 0`.

Reply:
Ok.